libsignal-protocol/tests/libsignal-protocol-c-tests.rs

```diff
@@ -313,7 +312,7 @@ fn test_basic_pre_key_v2() {
     // Have Alice process Bob's pre key bundle, which should fail due to a
     // missing unsigned pre key.
     let got = alice_session_builder.process_pre_key_bundle(&bob_pre_key_bundle);
-    assert_eq!(got, Err(InternalError::InvalidKey));
+    assert!(got.is_err());
```

Comment:
Can't this be

```suggestion
    assert_eq!(got, Err(Error::InvalidKey));
```

?  I would be a bit sad, knowing all errors are now all correctly typed, but you only check for is `is_err()`.

Reply:
So, the issue is the  new `Error` type can't derive `Eq` or `PartialEq` because `std::io::Error` and `std::time::SystemTimeError` don't. Comparing error enums by equality is mostly useful in tests. I have restored the real test with using a `match`, which works fine (see commit 3576a84).